ibis/backends/tests/test_sql.py

```diff
@@ -195,3 +195,14 @@ def test_union_aliasing(backend_name, snapshot):
     result = top_ten.union(bottom_ten)
 
     snapshot.assert_match(str(ibis.to_sql(result, dialect=backend_name)), "out.sql")
+
+
+def test_union_generates_predictable_aliases(con):
+    t = ibis.memtable(
+        data=[{"island": "Torgerson", "body_mass_g": 3750, "sex": "male"}]
+    )
+    sub1 = t.inner_join(t.view(), "island").mutate(island_right=lambda t: t.island)
```

Comment:
Actually you don't need to use `.view()` here anymore because of `dereference_sides()`

Reply:
Right, but that's the test case, so I'm going to leave it for now.